metrotranscom/doorway
# Storage interface

# Pull Request Template

## Issue Overview

This PR addresses #20 


## Description

- Create shared-services package with version 1.0.0
- Create a file service interface
- Create an implementation of the above interface which uses Cloudinary for storage and uses existing methods
- Refactor the above code so the return value of putFile() is the id generated by Cloudinary
- Change all existing callers to use this service instead of interacting directly with Cloudinary through existing methods
- Make small fixes to existing code to fix new errors found after the refactoring

Unit tests for the Cloudinary file service will be added in a later PR since I am still working on configuring Jest for the new package.

## How Can This Be Tested/Reviewed?

I added the Cloudinary keys locally and tested the listing photo upload functionality from the partners site. I was able to upload a photo for a new test listing, see the progress bar update and see the file was uploaded successfully.

![image](https://user-images.githubusercontent.com/67125998/228646442-90d8c38b-b96a-4652-8078-5318d62bf67c.png)

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
